components/CreateChain/MetadataForm.tsx

```diff
@@ -22,34 +24,76 @@ import { isSupportedNetwork, useWallet } from '@/web3';
 
 import { UploadImageForm } from '../UploadImageForm';
 
+const slugify = (str: string) =>
+  str
+    .toLowerCase()
+    .trim()
+    .replace(/[^\w\s-]/g, '')
+    .replace(/[\s_-]+/g, '-')
+    .replace(/^-+|-+$/g, '');
+
+const makeId = () => {
+  let result = '';
+  const characters = 'abcdefghijklmnopqrstuvwxyz0123456789';
+  const charactersLength = characters.length;
+  for (let i = 0; i < 6; i++) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+  }
+  return result;
+};
+
 export const MetadataForm: React.FC<{
   onBack?: () => void;
   onSubmit: (
     name: string,
     description: string,
     metadataUri: string,
+    slug?: string,
     imageUrl?: string,
   ) => void | Promise<void>;
 }> = ({ onBack, onSubmit }) => {
   const [nameRef, setName] = useInputText();
   const [descRef, setDescription] = useInputText();
+  const [slug, setSlug] = useState('');
+  const { getQuestChainsFromSlug } = graphql;
 
   const uploadImageProps = useDropImage();
   const { imageFile } = uploadImageProps;
 
   const { isConnected, chainId } = useWallet();
 
-  const isDisabled = !isConnected || !isSupportedNetwork(chainId);
+  const [slugAvailable, setSlugAvailable] = useState(true);
+
+  const isDisabled =
+    !isConnected ||
+    !isSupportedNetwork(chainId) ||
+    !slugAvailable ||
+    slug.match(/^[a-z0-9]+(?:-[a-z0-9]+)*$/) === null ||
+    ethers.utils.isAddress(slug);
 
   const [isSubmitting, setSubmitting] = useState(false);
 
+  const fetchSearchResults = async (slug: string) => {
+    if (chainId) {
+      const qcFromSlug = await getQuestChainsFromSlug(chainId, slug);
+      if (qcFromSlug.length === 0) {
+        setSlugAvailable(true);
+        setSlug(slug);
+      } else {
+        setSlugAvailable(false);
+        setSlug(`${slug}${makeId()}`);
```

Comment:
should we add a hyphen between the random id and the generated slug?

Reply:
that's a good idea, i'll add it!